### DISCUSSION TITLE
How to intercept routes from `/` page only?

### DISCUSSION THREAD

**Discussion by vovados1** [vercel/next.js#72037] [Help]
### Summary

Hi everyone! I'm trying to intercept the `/portfolio/project/[slug]` route from main page (`/`) ONLY. With the following structure I've already accomplished that (partially):

![image](https://github.com/user-attachments/assets/47e38c76-03b0-4783-a0ac-651ca13ef1c7)

The issue is that other routes that lead to that specific route also get intercepted which I don't want. For example, when I'm on `/portfolio` and click on one of the `/portfolio/project/[slug]` links, it gets intercepted and dialog window opens. Is there any native way to fix it? Thanks!

**Comment by aniruddhaadak80** [comment]
Hi @dx1ded ,

To intercept only the `/portfolio/project/[slug]` route from the main page (`/`), without affecting other navigation routes like `/portfolio`, you can utilize Next.js's route groups and conditional logic. Here's one way to approach it:

### Solution

1. **Check Current Path**: Use `useRouter` to determine if the current page is the root (`/`). If it is, intercept the navigation and show the dialog only when navigating from the root.

2. **Conditionally Render Dialog**: In your layout or component that controls the dialog, check if the `from` path is `/` before opening the dialog. This prevents the dialog from opening on internal navigations within `/portfolio`.

### Code Example

In your component, you can use `useRouter` as shown below:

```tsx
import { useRouter } from 'next/router';
import { useEffect } from 'react';

const InterceptDialog = ({ children }) => {
  const router = useRouter();

  useEffect(() => {
    const handleRouteChange = (url) => {
      if (router.pathname === '/' && url.startsWith('/portfolio/project/')) {
        // Open dialog logic here
      } else {
        // Close dialog if coming from other routes
      }
    };

    router.events.on('routeChangeStart', handleRouteChange);
    return () => {
      router.events.off('routeChangeStart', handleRouteChange);
    };
  }, [router]);

  return <>{children}</>;
};
```

This intercepts only when navigating from the main page (`/`). The dialog will open only if the route starts with `/portfolio/project/` from the root page and not from other pages. 

This solution is native to Next.js and allows you to control when the dialog appears based on the current route.

Thanks .
